include/pmacc/particles/ParticlesBase.kernel

```diff
@@ -916,16 +951,20 @@ struct KernelDeleteParticles
                 uint32_t const
             )
             {
-                /* all frames and particles are removed, set counter for particles
-                 * in the last frame to zero
-                 */
-                pb.getSuperCell( superCellIdx ).setSizeLastFrame( 0 );
+                // all frames and particles are removed
+                pb.getSuperCell( superCellIdx ).setNumParticles( 0 );
             }
         );
     }
 };
 
 /** copy particles from the guard to an exchange buffer
+ *
+ * @warning This kernel resets the number of particles in the processed supercells even
+ * if there are particles left in the supercell and not fulfill the rule that the
+ * last frame is contiguous filled.
+ * Call KernelFillGaps afterwards if you need a valid number of particles
+ * and a contiguous filled last frame.
```

Comment:
This comment seems to be mostly a copy of a comment at the header file, same fixes are probably needed here.